File: DecompiledGameScripts/cyberpunk/devices/security/coder.swift

public class Coder extends BasicDistractionDevice {

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as CoderController;
  }

  private const func GetController() -> ref<CoderController> {
    return this.m_controller as CoderController;
  }

  public const func GetDevicePS() -> ref<CoderControllerPS> {
    return this.GetController().GetPS();
  }
}

File: Definitions/Game_Definitions/classes/Coder.lua
---@meta
---@diagnostic disable

---@class Coder : BasicDistractionDevice
Coder = {}

---@return Coder
function Coder.new() return end

---@param props table
---@return Coder
function Coder.new(props) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Coder:OnTakeControl(ri) return end

---@return CoderController
function Coder:GetController() return end

---@return CoderControllerPS
function Coder:GetDevicePS() return end

